### PR TITLE
Re-organizing and extending tests

### PR DESCRIPTION
**Pull request by ljakimczuk**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: integration/test/validation/common.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+type appConfig struct {
+	appCatalog      string
+	appLabels       map[string]string
+	appName         string
+	appNamespace    string
+	appVersion      string
+	configName      string
+	inCluster       bool
+	targetCluster   string
+	targetNamespace string
```
Comment (rossf7) [inline integration/test/validation/common.go:29]: <3

**Review by rossf7 — COMMENTED** [review]
Direction is good just a few suggestions.

File: integration/test/validation/common.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+func createAppCatalog(ctx context.Context, name string) error {
+	// TODO: Remove once apptestctl creates catalog CRs.
+	catalogCR := &v1alpha1.Catalog{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "default",
+		},
+		Spec: v1alpha1.CatalogSpec{
+			Description: "This catalog holds Apps exclusively running on Giant Swarm control planes.",
```
[the remark this answers is not in the document]
Comment (ljakimczuk) [reply integration/test/validation/common.go]: Yeah, it so obvious after moving to commont and yet I haven't thought about it 🤦 😄

File: integration/test/validation/common.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+		}
+
+		return nil
+	}
+
+	return ensureCreated(ctx, o)
+}
+
+func createAppCatalog(ctx context.Context, name string) error {
+	// TODO: Remove once apptestctl creates catalog CRs.
```
[the remark this answers is not in the document]
Comment (ljakimczuk) [reply integration/test/validation/common.go]: Noted! Thanks!

File: integration/test/validation/common.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+	err := backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func executeWithApp(ctx context.Context, expectedError string, config appConfig) error {
```
[the remark this answers is not in the document]
Comment (ljakimczuk) [reply integration/test/validation/common.go]: Looks better, thanks!

File: integration/setup/config.go [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+	c := TestConfig{
+		AppTest:    appTest,
+		K8sClients: k8sClients,
+		Logger:     logger,
+	}
+
+	return c, nil
+}
+
+func (tc *TestConfig) CreateNamespace(ctx context.Context, name string) error {
```
Comment (rossf7) [inline integration/setup/config.go:82]: This is nicely structured and I like how you reuse the ensureExecuted method.

Just one nit. Can you order the methods alphabetically?

File: integration/test/mutation/app_test.go [hunk 6]
```diff
@@ -6,20 +6,16 @@ package mutation
...
-	corev1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-admission-controller/integration/helpers"
 )
 
 const (
-	appName        = "kiam-app"
+	appName        = "kaim-app"
```
Comment (rossf7) [inline integration/test/mutation/app_test.go:18]:
```suggestion
	appName        = "kiam-app"
```

File: integration/test/mutation/main_test.go [hunk 7]
```diff
@@ -4,78 +4,26 @@
...
-		err = appTest.InstallApps(ctx, apps)
-		if err != nil {
-			logger.Errorf(ctx, err, "install apps failed")
-			os.Exit(2)
-		}
-	}
-
-	os.Exit(m.Run())
+func TestMain(m *testing.M) {
+	setup.Setup(m, config)
```
Comment (rossf7) [inline integration/test/mutation/main_test.go:28]: Nice <3

**Review by rossf7 — APPROVED** [review]
Nice, LGTM

Just a few minor nits.

File: integration/setup/config.go (hunk 5 above)
Comment (ljakimczuk) [reply integration/setup/config.go:82]: sure! It's done

File: integration/test/mutation/app_test.go (hunk 6 above)
Comment (ljakimczuk) [reply integration/test/mutation/app_test.go]: Done

**Pull request merged**